p2p/p2p.go

```diff
@@ -230,6 +232,11 @@ func (s *Service) Host() host.Host {
 	return s.host
 }
 
+// MsgChan returns a channel where all messages that arrive are sent
+func (s *Service) MsgChan() chan Message {
```

Comment:
I don't think this should be exported

Reply:
how will we get the messages out of p2p?

Reply:
I can see now we simply have different ideas for how this channel should be shared. My inclination was to pass a channel into both services at startup (ie. as part of the constructor). I think this will be okay as long as it is a uni-directional channel -- please update above if you want to stick with this.

Reply:
I see, I'll change it then. it's read-only within the core service already

Reply:
I've changed it to initialize a bidirectional channel in cmd/configcmd.go which is then passed in to p2p.Service and core.Service, where it is write-only and read-only respectively. let me know if it looks reasonable.

btw I couldn't put the msgChan in the p2p config since it messed up the cmd tests (chan can't be marshalled to toml) so I just passed it in through the constructor instead